deepcell/data/tracking_test.py

```diff
@@ -0,0 +1,180 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for tracking functions"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.testing import parameterized
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow.python.platform import test
+
+from deepcell.data import tracking
+
+
+@parameterized.named_parameters([
+    {
+        'testcase_name': 'test_data_01',
+        'time': 30,
+        'max_cells': 25,
+        'crop_size': 32,
+        'batch_size': 32,
+        'seed': None,
+        'track_length': 8,
+        'val_split': 0.4
+    }, {
+        'testcase_name': 'test_data_02',
+        'time': 40,
+        'max_cells': 29,
+        'crop_size': 32,
+        'batch_size': 24,
+        'seed': 2,
+        'track_length': 7,
+        'val_split': 0.2
+    }, {
+        'testcase_name': 'test_data_03',
+        'time': 50,
+        'max_cells': 15,
+        'crop_size': 32,
+        'batch_size': 10,
+        'seed': 4,
+        'track_length': 6,
+        'val_split': 0.1
+    }
+])
+class TrackingTests(test.TestCase, parameterized.TestCase):
+
+    def test_temporal_slice(self, time, max_cells, crop_size,
+                            batch_size, seed, track_length, val_split):
+        X, y = self.create_test_data(time, max_cells, crop_size)
+        sliced_X, sliced_y = tracking.temporal_slice(X, y, track_length)
+
+        # Test temporal dimension correct index and value
+        for key in sliced_X:
+            self.assertEqual(sliced_X[key].shape[0], track_length)
+        for key in sliced_y:
+            self.assertEqual(sliced_y[key].shape[0], track_length - 1)
+
+    def test_random_rotate(self, time, max_cells, crop_size,
+                           batch_size, seed, track_length, val_split):
+        X, y = self.create_test_data(time, max_cells, crop_size)
+
+        # Get appearance and centroid tensors
+        X_apps = X['appearances']
+        X_cents = X['centroids']
+
+        # Assert rotating does not mutate the shape of tensors
+        rotated_X, y = tracking.random_rotate(X, y, 180)
+        self.assertEqual(rotated_X['appearances'].shape, X_apps.shape)
+        self.assertEqual(rotated_X['centroids'].shape, X_cents.shape)
+
+        # Assert rotating by 0 does not change tensors
+        X_apps = rotated_X['appearances'][0, 0, 0, 0]
+        X_cents = rotated_X['centroids'][0, 0, 0]
+        output_X, y = tracking.random_rotate(rotated_X, y, 0)
+        r0 = (float)(output_X['appearances'][0, 0, 0, 0] - X_apps)
+        self.assertEqual(r0, 0)
+        r0 = (float)(output_X['centroids'][0, 0, 0] - X_cents)
+        self.assertEqual(r0, 0)
```

Comment:
It looks like you are testing that a single value of the X arrays are unchanged.

Instead, check that the entire array is the same with the [`assertAllEqual` testing helper](https://www.tensorflow.org/api_docs/python/tf/test/TestCase).

Something like:

```python
X_apps = rotated_X['appearances']
X_cents = rotated_X['centroids']

output_X, y = tracking.random_rotate(rotated_X, y, 0)

self.assertAllEqual(output_X['appearances'], X_apps)
self.assertAllEqual(output_X['centroids'], X_cents)
```